src/memory.rs

```diff
@@ -55,6 +55,12 @@ impl Pointer {
     pub fn points_to_zst(&self) -> bool {
         self.alloc_id == ZST_ALLOC_ID
     }
+    pub fn from_int(i: usize) -> Self {
+        Pointer {
+            alloc_id: ZST_ALLOC_ID,
```

Comment:
Okay, this is clever. Although it effectively makes ZST allocations the same as null pointers, but `&T` can't be null.